File: php/features.php
<?php
declare(strict_types=1);

// JsonIpGeolocation SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class JsonIpGeolocationFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new JsonIpGeolocationBaseFeature();
            case "test":
                return new JsonIpGeolocationTestFeature();
            default:
                return new JsonIpGeolocationBaseFeature();
        }
    }
}

File: rb/features.rb
# JsonIpGeolocation SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/test_feature'


module JsonIpGeolocationFeatures
  def self.make_feature(name)
    case name
    when "base"
      JsonIpGeolocationBaseFeature.new
    when "test"
      JsonIpGeolocationTestFeature.new
    else
      JsonIpGeolocationBaseFeature.new
    end
  end
end
